bin/lib/rust_library_builder.py

```diff
@@ -0,0 +1,453 @@
+import contextlib
```

Comment:
As this is mostly a copy (I think) of existing stuff, I won't go into deep analysis here to save a bunch of "yeah but that's how it is in the other file"